packages/itwin/measure-tools/src/measurements/DistanceMeasurement.ts

```diff
@@ -95,13 +96,16 @@ export class DistanceMeasurement extends Measurement {
   private _startPoint: Point3d;
   private _endPoint: Point3d;
   private _showAxes: boolean;
+  private _measureLine?: Point3d[];
```

Comment:
You're going to have problems with these new fields not being persisted in the wire-props. The measurement, if persisted, won't come back with the same state.

Also, in general I'm a bit confused by both measureLine and secondaryLine. Like in the case of MeasureHeight, it's the same line? And measureLine is overriding the start/end points when drawing, really the start/stop is the blue hypotenuse the user is drawing in the gifs.

TBH it sounds like these tools need their own subclass of the toolmodel to modify the start/stop points being fed to the dynamic measurement, and then the distance measurement just having an optional "leader line(s)" to relate the measurement to some point it was referenced from during construction, which is what I think you're going for with these new tools.

Reply:
Resolving this comment as the approach has changed quite a bit since.